### DISCUSSION TITLE
如何在无监督任务中使用ColossalAI

### DISCUSSION THREAD

**Discussion by pigen01** [hpcaitech/ColossalAI#1065] [Community | Q&A]
无监督行人重识别任务中, 会出现如下将特征聚类中心写进分类器权值的操作,在ColossalAI中该如何实现?
`
model.module.classifier.weight.data[0:source_classes].copy_(source_centers.cuda())
`

**Comment by pigen01** [comment]
在 deepspeed 中可以通过 GatheredParameters 回收某一层的权重参数，colossalAI 中是否也存在类似的函数？如果有的话该如何实现？
```
self.position_embeddings = torch.nn.Embedding(...)
with deepspeed.zero.GatheredParameters(self.position_embeddings.weight, modifier_rank=0):
    # Initialize the position embeddings.
    self.init_method(self.position_embeddings.weight)
```

**Reply by ver217** [reply]
暂时没有提供类似的API，目前可以通过load_state_dict间接完成这个功能
